integration-test/src/test/java/uk/gov/gchq/gaffer/integration/impl/loader/AbstractLoaderIT.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.integration.impl.loader;
+
+import com.google.common.collect.Iterables;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import uk.gov.gchq.gaffer.commonutil.TestGroups;
+import uk.gov.gchq.gaffer.commonutil.TestPropertyNames;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.data.element.id.EdgeId;
+import uk.gov.gchq.gaffer.data.element.id.EntityId;
+import uk.gov.gchq.gaffer.integration.AbstractStoreIT;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.store.schema.TestSchemas;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static junit.framework.TestCase.assertEquals;
+import static uk.gov.gchq.gaffer.data.util.ElementUtil.assertElementEquals;
+
+public abstract class AbstractLoaderIT<T extends Operation> extends AbstractStoreIT {
+
+    protected final Iterable<? extends Element> input = getInputElements();
+
+    @Override
+    public void setup() throws Exception {
+        super.setup();
+        configure(input);
+    }
+
+    @Test
+    public void shouldAddElements_basicSchema() throws OperationException {
+        // Given
+        createGraph(TestSchemas.getBasicSchema());
+
+        // When
+        addElements();
+        final Iterable<? extends Element> result = getAllElements();
+
+        // Then
+        assertEquals(Iterables.size(input), Iterables.size(result));
```

Comment:
I think we need some more verifications to check the data is loaded in correctly, I think we discussed doing a seeded lookup (GetElements) and checking that works to ensure the vertices have been indexed correctly. It would also be worth testing all of the different filters and flags. Similar to what we currently do in GetElementsIT, FilteringIT etc.